src/core/utils/terminal/terminal.cc

```diff
@@ -157,7 +157,7 @@ bool Terminal::unicode_allowed() const noexcept {
 
 TerminalSize Terminal::get_size() {
   #if DT_OS_WINDOWS
-    _detect_window_size();
+    if (!is_plain_) _detect_window_size();
   #else
     if (!size_.width || !size_.height) _detect_window_size();
   #endif
```

Comment:
Can merge the two branches together:
```suggestion
  if (!is_plain_ &&
      (DT_OS_WINDOWS || size_.width == 0)) _detect_window_size();
```

Reply:
Sure, this part is not performance critical, so won't loose much.